core-domain/src/test/java/org/uniprot/core/uniprotkb/impl/UniProtKBEntryBuilderTest.java

```diff
@@ -140,6 +143,51 @@ void canSet_internalSection() {
         assertEquals(internalSection, entry.getInternalSection());
     }
 
+    @Test
+    void cannotAddExtraAttributeWithNullValue() {
+        UniProtKBEntry entry =
+                UniProtKBEntryBuilder.from(minEntry).extraAttributesAdd("attrib1", null).build();
+        assertNotNull(entry);
+        assertEquals(0, entry.getExtraAttributes().size());
+    }
+
+    @Test
+    void cannotAddExtraAttributeWithEmptyKey() {
+        UniProtKBEntry entry =
+                UniProtKBEntryBuilder.from(minEntry).extraAttributesAdd("", "value1").build();
+        assertNotNull(entry);
+        assertEquals(0, entry.getExtraAttributes().size());
+    }
+
+    @Test
+    void canAddExtraAttributes() {
+        UniProtKBEntry entry =
+                UniProtKBEntryBuilder.from(minEntry)
+                        .extraAttributesAdd("attrib1", "value1")
+                        .build();
+        assertNotNull(entry);
+        assertEquals(1, entry.getExtraAttributes().size());
+        assertTrue(entry.getExtraAttributes().containsKey("attrib1"));
+        assertEquals("value1", entry.getExtraAttributes().get("attrib1"));
+    }
+
```

Comment:
can we add another test to show what is supposed to happen when someone adds the same key two times?
```java
                UniProtKBEntryBuilder.from(minEntry)
                        .extraAttributesAdd("attrib1", "value1")
                        .extraAttributesAdd("attrib1", "value2")
                        .build();
// what is the value of "attrib1" in getExtraAttributes()?
```